### DISCUSSION TITLE
Example for using buttons

### DISCUSSION THREAD

**Discussion by beathoch** [siku2/hass-dingz#49] [General]
Hi,
The dingz Home Assistant Integration works very well in principle. You can see all sensors, etc.

As specified in the instructions, it is necessary to install the MQTT broker using the dingz buttons. This also worked very well and with the MQTT Explorer I see all events.

I think the base is created. But how can you use the dingz buttons in Home Assistant now?
Do you have to create a device for the events in configuration.yaml? or do you have to take on the events when creating an automation?

How do you deal with that? can anyone please show an example?

I'm not the only one who's on this point. It is completely unclear how to use the Events buttons in the automations.

thank you very much for your great job! this integration is not only for me very important.

**Comment by PackElend** [comment]
I would like to know the same thing. Any progress in this matter. Setting it up in configuration.yaml seems to be kind of nightmare

**Reply by siku2** [reply]
Could you take some time to describe what you're trying to achieve?
For what it's worth, I fully agree that the user experience for buttons could be much better. As it stands the integration simply forwards the internal events emitted by the dingz device, which aren't designed to be user-friendly.

**Comment by PackElend** [comment]
I can certainly share more details.
I’m new to Home Assistant. I’m trying to create an automation that will trigger a Shelly Wave Shutter (https://www.shelly.com/products/shelly-qubino-wave-shutter?srsltid=AfmBOopoRDQTCz-6U1e9XtlHskynVlozKG3DZZrTKa0T8urediWkJkVF) when a button on a Dingz (https://www.dingz.ch/?r=0) is pressed.

It sounds like a simple thing, but setting up the event entity via the GUI feels like rocket science! 
Searching MQTT Event in [Integrations - Home Assistant](https://www.home-assistant.io/integrations/?search=mqtt+event) results in [MQTT Event - Home Assistant](https://www.home-assistant.io/integrations/event.mqtt/) but there is noting in like this in the integration 
<img width="423" height="541" alt="image" src="https://github.com/user-attachments/assets/51374c6f-947d-44f3-a7f1-d4fbdbfd8c84" />

**Reply by siku2** [reply]
Just to confirm, did you set up the MQTT part of this integration as documented here: https://github.com/siku2/hass-dingz/blob/main/docs/mqtt.md ?

Button events can only work over MQTT.

**Reply by PackElend** [reply]
I use mosquitto + home assistant mqqt integration.
If I carry out "listen" it reads the same as shown in mqtt Explorer.
I can share more details tomorrow

**Reply by PackElend** [reply]
1. Broker: [Eclipse Mosquitto](https://mosquitto.org/)
2. Verfication: [MQTT Explorer | An all-round MQTT client that provides a structured topic overview](https://mqtt-explorer.com/)
3. Home Assistant integration: [MQTT - Home Assistant](https://www.home-assistant.io/integrations/mqtt/)
   <img width="540" height="279" alt="image" src="https://github.com/user-attachments/assets/8634265e-6859-463c-8c62-070cbdae0c22" />

proof of communcation:
1. reading the broker with MQTT Explorer
    <img width="562" height="292" alt="image" src="https://github.com/user-attachments/assets/599aecd8-a889-4b8d-8e65-0adcf919aa82" />
1. MQTT integration: listen to user dcefined topic 
   <img width="1063" height="432" alt="image" src="https://github.com/user-attachments/assets/99e4dda9-4c48-4143-abe5-2b0a58461d99" />

**Reply by siku2** [reply]
Okay, and if you open up one of the devices in HA (see https://my.home-assistant.io/redirect/integration/?domain=dingz) do you see any entities under "Events" like this?

<img width="338" height="405" alt="image" src="https://github.com/user-attachments/assets/7580a700-208f-43cb-9d0f-e5843f0c58ec" />

These (apart from the PIR) are your buttons.

**Reply by PackElend** [reply]
> https://my.home-assistant.io/redirect/integration/?domain=dingz)

Forwards me to
<img src="https://github.com/user-attachments/assets/25f30739-828d-433d-958d-1be8fbc85db3" width=25%>

Here are the details you asked as shown in the HA Android app 
<img src="https://github.com/user-attachments/assets/0ea1fd09-11db-4be0-ab72-e2b16e601552" width=25%>

**Reply by siku2** [reply]
~~Yes, now just press on one of the two devices :)
I can't create a link directly to one of your devices because I obviously don't know what devices you have.~~

EDIT: I see you added more screenshots. Yes, those events are the 4 buttons + the PIR sensor. You should notice a change when you physically press one of the buttons.
The events emitted are a bit obscure, you can see their meaning here:

https://github.com/siku2/hass-dingz/blob/7aad46a2904799cc518ccb57d33d557414348099/custom_components/dingz/translations/en.json#L59-L66

Now you can just set up an automation based on those button entities. See the official docs for reference: https://www.home-assistant.io/integrations/event

**Reply by PackElend** [reply]
> You should notice a change when you physically press one of the buttons.

Confirmed, time since event is much lower now:
<img src="https://github.com/user-attachments/assets/bcae847d-562c-4482-a089-e437a3050b3c " width=50%>
